src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala

```diff
@@ -192,6 +192,38 @@ class MarathonHealthCheckManager @Inject() (
       Future.sequence(taskHealth)
     }
 
+  override def healthCounts(appId: PathId): Future[HealthCounts] =
+    withReadLock {
+      implicit val timeout: Timeout = Timeout(2, SECONDS)
+      val tasks = taskTracker.get(appId)
+
+      // aggregate health check results of each tasks to
+      // -1: at least one task unhealthy
+      // 0: at least one task unknown, none unhealthy
+      // 1: all tasks healthy
```

Comment:
Yes, but I think the pure numbers are easier to understand in this case because the order of the numbers if what matters, not symbolic names. Compare the list.min some lines below.